eng/scripts/generate_from_source_pom.py

```diff
@@ -55,7 +56,7 @@
 maven_xml_namespace = '{http://maven.apache.org/POM/4.0.0}'
 
 # Function that creates the aggregate POM.
-def create_from_source_pom(project_list: str):
+def create_from_source_pom(project_list: str, set_pipeline_variable: bool = False):
```

Comment:
Instead of this being a boolean I suggest that it be the name of the variable. It would allow opt-in and customization of the specific variable.

Reply:
I'm not quite sure what you mean? Do you mean not setting a type hint+default value?

Reply:
I'm suggesting we make this parameter a string that called pipeline_variable that we can pass in the value CheckoutDirectories in our case. If the variable is set then we set it below if not then we don't. That makes sure that we can keep the name of this variable consistent in our different usages all in the yml file and not have these as tightly coupled.